### ISSUE TITLE
dotnet command missing (at least on mac)

### ISSUE THREAD

**Issue by mikehaas763** [dotnet/cli#771]
I've already had dotnet installed through the mac .pkg install. It's been a while since I downloaded it so I went and downloaded a new copy and installed it to get updated with the latest.

Now I have no `dotnet` command available. I've made sure that I've opened a new shell make sure it wasn't that.

**Comment by piotrpMSFT** [comment]
Whoa, interesting. @blackdwarf can you take a look?

**Comment by blackdwarf** [comment]
@mikehaas763 so you've installed the PKG a while back, then you've downloaded and installed a recent one (you didn't remove the old one) and now there is no `dotnet` command?

**Comment by mikehaas763** [comment]
@blackdwarf exactly. I also tried redownloading the PKG several times and reinstalling. I'm not sure how to uninstall so I haven't attempted to try.

**Comment by blackdwarf** [comment]
@mikehaas763 uninstalling the PKG is a bit involved. There is a good [SuperUser question](http://superuser.com/questions/36567/how-do-i-uninstall-any-apple-pkg-package-file) that can help you with that. This is just FYI, of course.

Can you check if /usr/local/share/dotnet exists and what is in it? Also, can you do a `ls -lhF /usr/local/bin/dotnet` and add the output to this issue? Thanks!

**Comment by mikehaas763** [comment]
```
mufasa:~ mike$ ls -l /usr/local/share/dotnet/
total 0
drwxr-xr-x  225 root  wheel   7.5K Jan  9 13:54 bin/
drwxr-xr-x    2 mike  staff    68B Jan  9 13:52 cli/
drwxr-xr-x    3 root  wheel   102B Jan  9 12:17 runtime/
```

```
mufasa:~ mike$ ls -lhF /usr/local/bin/dotnet
lrwxr-xr-x  1 root  admin    38B Nov 21 20:49 /usr/local/bin/dotnet@ -> /usr/local/share/dotnet/cli/bin/dotnet
```

So looks like there is a symlink in the standard $PATH but it's pointing to a file in the cli dir when bin is a sibling directory of cli. Is that what the path used to be? Maybe the installer needs to not just check for the existence of the symlink but of the file it points to? I'm just making an assumption there because I have no idea what the PKG internals actually does.

For my specific case I'll fix my symlink.

**Comment by blackdwarf** [comment]
@mikehaas763 yes, that is the problem. The main issue is that you've installed a newer PKG over a very old one. The PKG should recreate all of the symlinks, but for whatever reason it didn't. I tried this today (installing PKG twice without removing anything in between installs) and it worked correctly for me. 

So, yes, some more work should be done in the postinstall (or maybe preinstall) steps to ensure cleanup. Maybe just removing all of the symlinks and then recreating them.

**Comment by piotrpMSFT** [comment]
Fixed by #840
